packages/sitecore-jss-react/src/enhancers/withSitecoreContext.tsx

```diff
@@ -1,27 +1,31 @@
-import React, { ReactNode } from 'react';
+import React from 'react';
+import { EnhancedOmit } from '@sitecore-jss/sitecore-jss/utils';
 import { SitecoreContextReactContext, SitecoreContextValue } from '../components/SitecoreContext';
 
 export interface WithSitecoreContextOptions {
   updatable?: boolean;
 }
 
+// The props that HOC will inject
 export interface WithSitecoreContextProps {
-  sitecoreContext?: SitecoreContextValue;
+  sitecoreContext: SitecoreContextValue;
   updateSitecoreContext?: ((value: SitecoreContextValue) => void) | false;
 }
 
-export interface ComponentConsumerProps extends WithSitecoreContextProps {
-  children?: ReactNode;
-}
+// The props that HOC will receive.
+export type WithSitecoreContextHocProps<ComponentProps> = EnhancedOmit<
```

Comment:
did you test this with Omit? Can't we achieve the same result with the Omit type?

Reply:
It works in the same way like the old [implementation](https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss-react/src/enhancers/withSitecoreContext.tsx#L17).
The difference between new _EnhancedOmit_ type and default _Omit_ you can see [here](https://github.com/Sitecore/jss/pull/1797/files/ac621f6eadacb409b6cb937cc50f0c1ef844976a#diff-9828ba0e7ecd88bab97f6329714060a541c5808eece1b938506fe0f7e9695989R11)
In our case default _Omit_ doesn't work because we use _withSitecoreContext_ in Placeholder component, which has [indexing type](https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss-react/src/components/PlaceholderCommon.tsx#L31)